lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -591,6 +591,355 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
+                centerFreqTensorL2CZero.getType()),
+            ValueRange{centerFreqTensorL2CZero, maskLowToCenterInt});
+        Value oneConstTensor = Torch::createRank0Tensor(
+            rewriter, binder.getLoc(), VTType::get(ctx, std::nullopt, f32Ty),
+            oneConst);
+        Value upslopeL2C = b.create<Torch::AtenIndexPutOp>(
+            zeroToNumElesType, upslopeL2C1, upslopeOneIdxList, oneConstTensor,
+            falseConst);
+
+        // H2C computation
+        Value maskCenterToHighZero =
+            b.create<Torch::AtenEqScalarOp>(maskType, centerToHigh, zeroConst);
+        Value maskH2CBeforeC = b.create<Torch::AtenLtTensorOp>(
+            maskL2CAfterCType, zeroToNumEles, centerFreqTensor);
+        Value centerToHighNoZero = b.create<Torch::AtenWhereScalarSelfOp>(
+            unsqueezeResType, maskCenterToHighZero, negOneConst, centerToHigh);
+        Value c2hNZFlt = b.create<Torch::AtenToDtypeOp>(
+            l2cNZFltTy, centerToHighNoZero, /*dtype=*/sixConst,
+            /*non_blocking=*/falseConst, /*copy=*/falseConst,
+            /*memory_format=*/noneConst);
+        Value zeroToNumElesC2H = b.create<Torch::AtenWhereScalarSelfOp>(
+            zeroToNumElesType, maskCenterToHighZero, zeroConst, zeroToNumEles);
+        Value downslopeDiff = b.create<Torch::AtenSubTensorOp>(
+            zeroToNumElesType, highFreqTensor, zeroToNumElesC2H,
+            /*alpha=*/oneConst);
+        Value downslopeC2H0 = b.create<Torch::AtenDivTensorOp>(
+            zeroToNumElesType, downslopeDiff, c2hNZFlt);
+        Value maskDownslopeC2H0Pos = b.create<Torch::AtenGtScalarOp>(
+            maskUpslopeL2C0PosType, downslopeC2H0, zeroConst);
+        Value downslopeC2H0Pos = b.create<Torch::AtenWhereScalarOtherOp>(
+            zeroToNumElesType, maskDownslopeC2H0Pos, downslopeC2H0, zeroConst);
+        Value idxH2CBeforeCList = b.create<Torch::PrimListConstructOp>(
+            rewriter.getType<Torch::ListType>(maskH2CBeforeC.getType()),
+            ValueRange{maskH2CBeforeC});
+        Value downslopeC2H = b.create<Torch::AtenIndexPutOp>(
+            zeroToNumElesType, downslopeC2H0Pos, idxH2CBeforeCList,
+            zeroConstTensor, falseConst);
+
+        // final result Calculation
+        Value maskH2CNonZero = b.create<Torch::AtenNeScalarOp>(
+            maskL2CAfterCType, downslopeC2H, zeroConst);
+        Value idxH2CNZList = b.create<Torch::PrimListConstructOp>(
+            rewriter.getType<Torch::ListType>(maskH2CNonZero.getType()),
+            ValueRange{maskH2CNonZero});
+        Value upslopeL2CMasked = b.create<Torch::AtenIndexPutOp>(
+            zeroToNumElesType, upslopeL2C, idxH2CNZList, zeroConstTensor,
+            falseConst);
+
+        Value slopesFinal = b.create<Torch::AtenAddTensorOp>(
+            zeroToNumElesType, upslopeL2CMasked, downslopeC2H,
+            /*alpha=*/oneConst);
+        // TODO: Set the right output type as given in attr
```

Comment:
Please add this support since there's no point in generating a result with wrong dtype and should't be a big thing to do.